nitransforms/tests/test_nonlinear.py

```diff
@@ -19,34 +21,88 @@
 }
 
 
+@pytest.mark.parametrize('size', [(20, 20, 20), (20, 20, 20, 3)])
+def test_itk_disp_load(size):
+    """Checks field sizes."""
+    with pytest.raises(TransformFileError):
+        ITKDisplacementsField.from_image(
+            nb.Nifti1Image(np.zeros(size), None, None))
+
+
+@pytest.mark.parametrize('size', [(20, 20, 20), (20, 20, 20, 1, 3)])
+def test_displacements_bad_sizes(size):
+    """Checks field sizes."""
+    with pytest.raises(ValueError):
+        DisplacementsFieldTransform(
+            nb.Nifti1Image(np.zeros(size), None, None))
+
+
+def test_itk_disp_load_intent():
+    """Checks whether the NIfTI intent is fixed."""
+    with pytest.warns(UserWarning):
+        field = ITKDisplacementsField.from_image(
+            nb.Nifti1Image(np.zeros((20, 20, 20, 1, 3)), None, None))
+
+    assert field.header.get_intent()[0] == 'vector'
+
+
+@pytest.mark.parametrize('image_orientation', ['RAS', 'LAS', 'LPS', 'oblique'])
 @pytest.mark.parametrize('sw_tool', ['itk'])
-def test_displacements_field(tmp_path, data_path, sw_tool):
+@pytest.mark.parametrize('axis', [0, 1, 2, (0, 1), (1, 2), (0, 1, 2)])
+def test_displacements_field1(tmp_path, get_testdata, image_orientation, sw_tool, axis):
+    """Check a translation-only field on one or more axes, different image orientations."""
     os.chdir(str(tmp_path))
```

Comment:
unrelated but caught it now
```suggestion
    tmp_path.chdir()
```

Reply:
`tmp_path` doesn't have `chdir()`